plugins/cake/cake_tin_delay_

```diff
@@ -0,0 +1,377 @@
+		[ $has_if_in -eq 0 ] && echo "video.negative video_in"
+		
+		if [ $has_if_in -eq 0 ]; then
+			echo "voice_in.label Voice"
+			echo "voice_in.type GAUGE"
+			echo "voice_in.min 0"
+			echo "voice_in.info avg delay for Voice packets thru cake"
+			echo "voice_in.graph no"
+		fi
+		echo "voice.label Voice"
+		echo "voice.type GAUGE"
+		echo "voice.min 0"
+		echo "voice.info avg delay for Voice packets thru cake"
+		[ $has_if_in -eq 0 ] && echo "voice.negative voice_in"
+		
+		exit 0
+	fi
+	
+	if [ "$diffserv_no" == "8" ]; then
+		
+		if [ $has_if_in -eq 0 ]; then
+			echo "tin1_in.label Tin 1"
+			echo "tin1_in.type GAUGE"
+			echo "tin1_in.min 0"
+			echo "tin1_in.info avg delay for Tin 1 packets thru cake"
+			echo "tin1_in.graph no"
+		fi
+		echo "tin1.label Tin 1"
+		echo "tin1.type GAUGE"
+		echo "tin1.min 0"
+		echo "tin1.info avg delay for Tin 1 packets thru cake"
+		[ $has_if_in -eq 0 ] && echo "tin1.negative tin1_in"
+		
+		for i in `seq 2 8`; do
+			if [ $has_if_in -eq 0 ]; then
+				echo "tin${i}_in.label Tin $i"
+				echo "tin${i}_in.type GAUGE"
+				echo "tin${i}_in.min 0"
+				echo "tin${i}_in.info avg delay for Tin $i packets thru cake"
+				echo "tin${i}_in.graph no"
+			fi
+			echo "tin${i}.label Tin $i"
+			echo "tin${i}.type GAUGE"
+			echo "tin${i}.min 0"
+			echo "tin${i}.info avg delay for Tin $i packets thru cake"
+			[ $has_if_in -eq 0 ] && echo "tin${i}.negative tin${i}_in"
+		done
+		
+		exit 0  
+	fi
+	
+	echo "cake_tin_delay_: no diffserv or unknown parameter set (best effort?)" >&2
+	exit 1
+	
+	;;
+esac
+
+if [ "$diffserv_no" == "3" ]; then
+	tc_cake_get_delays "$DEVICE" | awk '{
+	print "bulk.value " $1
+	print "besteffort.value " $2
+	print "voice.value " $3
+	}'
+	if [ $has_if_in -eq 0 ]; then
+		tc_cake_get_delays "$DEVICE_IN" | awk '{
+		print "bulk_in.value " $1
+		print "besteffort_in.value " $2
+		print "voice_in.value " $3
+		}'
+	fi
+	exit 0
+fi
+if [ "$diffserv_no" == "4" ]; then
+	tc_cake_get_delays "$DEVICE" | awk '{
+	print "bulk.value " $1
+	print "besteffort.value " $2
+	print "video.value " $3
+	print "voice.value " $4
+	}'
+	if [ $has_if_in -eq 0 ]; then
+		tc_cake_get_delays "$DEVICE_IN" | awk '{
+		print "bulk_in.value " $1
+		print "besteffort_in.value " $2
+		print "video_in.value " $3
+		print "voice_in.value " $4
+		}'
+	fi
+	exit 0
+fi
+if [ "$diffserv_no" == "8" ]; then
+	tc_cake_get_delays "$DEVICE" | awk '{
+	print "tin1.value " $1
+	print "tin2.value " $2
+	print "tin3.value " $3
+	print "tin4.value " $4
+	print "tin5.value " $5
+	print "tin6.value " $6
+	print "tin7.value " $7
+	print "tin8.value " $8
+	}'
+	if [ $has_if_in -eq 0 ]; then
+		tc_cake_get_delays "$DEVICE_IN" | awk '{
+		print "tin1_in.value " $1
+		print "tin2_in.value " $2
+		print "tin3_in.value " $3
+		print "tin4_in.value " $4
+		print "tin5_in.value " $5
+		print "tin6_in.value " $6
+		print "tin7_in.value " $7
+		print "tin8_in.value " $8
+		}'
+	fi
+	exit 0
+fi
+
+
```

Comment:
Please add an error message instead of silently failing.

Reply:
alright